config/dev.exs

```diff
@@ -87,7 +87,9 @@ config :archethic, Archethic.OracleChain.Scheduler,
   # Aggregate chain at the 50th second
   summary_interval: "0 * * * * *"
 
-config :archethic, Archethic.Networking.IPLookup, Archethic.Networking.IPLookup.Static
+config :archethic, Archethic.Networking.IPLookup,
+  provider: Archethic.Networking.IPLookup.Static,
+  ipify_provider: Archethic.Networking.IPLookup.IPIFY
```

Comment:
Why do you want to add this ipify provider conf ?

Reply:
If you want to leverage mock for testing, you might define IPIFY or NAT as a behavior, where a default implementation will be the NAT or IPIFY and the test impl the mock

Reply:
> If you want to leverage mock for testing, you might define Public IP gateway or Local Discovery as a behavior, where a default implementation of the public gateway could be IPIFY and during testing the Mock and the local discovery default impl being NAT

Hello.
Are you referring to  default_implmentation like this 

```
defmodule Archethic.Networking.IPLookup.PublicIPGatewayImpl do
  @moduledoc false

  defmacro __using__(_) do
    quote do
      def get_node_ip() do
        Logger.info("Trying IPFY as fallback")
        Archethic.Networking.IPLookup.IPIFY.get_node_ip()
        defoverridable get_node_ip: 0
      end
    end
  end

  @callback get_node_ip() :: {:ok, :inet.ip_address()} | {:error, any()}
end
```
either -- 
```
defmodule Archethic.Networking.IPLookup.PublicIPGatewayImpl do
  @moduledoc false
  @callback get_node_ip() :: {:ok, :inet.ip_address()} | {:error, any()}
end
```
```
defmodule Archethic.Networking.IPLookup.IPIFY do
  @moduledoc """
  Module provides external IP address of the node identified by IPIFY service.
  """

  alias Archethic.Networking.IPLookup.PublicIPGatewayImpl

  @behaviour PublicIPGatewayImpl

  @impl PublicIPGatewayImpl
  @spec get_node_ip() :: {:ok, :inet.ip_address()} | {:error, :not_recognizable_ip}
  def get_node_ip do
    with {:ok, {_, _, inet_addr}} <- :httpc.request('http://api.ipify.org'),
         {:ok, ip} <- :inet.parse_address(inet_addr) do
      {:ok, ip}
    else
      {:error, {:failed_connect, _reason}} ->
        {:error, :not_recognizable_ip}

      {:error, :einval} ->
        :inets.stop()
        {:error, :not_recognizable_ip}
    end
  end
end
```
either ---
Is it correct to have PublicIP Module, in which implements IPify to get public IP?

Reply:
Nice you have tried with Macro 😉 

But you can do much more simplifier using Knigge

```elixir
defmodule Archethic.Networking.IPLookup.PublicGateway do
    use Knigge, otp_app: :archethic, default:  __MODULE__.IPIFYImpl

    @callback get_ip() :: {:ok, :inet.ip_address()} | {:error, :not_recognizable_ip}
end
```

This will create, at compile time, the function PublicGateway.get_ip() pointing to IPIFYImpl

But this can be override for test as:
```elixir
config :archethic, Archethic.Networking.IPLookup.PublicGateway, MockPublicGateway
```

Reply:
Hello.
I missed you review, 
Will do the changes now.